clients/lsp-tailwindcss.el

```diff
@@ -0,0 +1,212 @@
+  "Unknown variant name used with the
+  @variants directive(https://tailwindcss.com/docs/functions-and-directives/#variants)"
+  :type '(choice
+          (const "ignore")
+          (const "warning")
+          (const "error"))
+  :package-version '(lsp-mode . "7.1"))
+
+(defcustom lsp-tailwindcss-lint-invalid-config-path "error"
+  "Unknown or invalid path used with the
+  theme helper(https://tailwindcss.com/docs/functions-and-directives/#theme)"
+  :type '(choice
+          (const "ignore")
+          (const "warning")
+          (const "error"))
+  :package-version '(lsp-mode . "7.1"))
+
+(defcustom lsp-tailwindcss-lint-invalid-tailwind-directive "error"
+  "Unknown value used with the
+  @tailwind directive(https://tailwindcss.com/docs/functions-and-directives/#tailwind)"
+  :type '(choice
+          (const "ignore")
+          (const "warning")
+          (const "error"))
+  :package-version '(lsp-mode . "7.1"))
+
+(defcustom lsp-tailwindcss-experimental-class-regex '()
+  ""
+  :type '(repeat string)
+  :package-version '(lsp-mode . "7.1"))
+
+(defcustom lsp-tailwindcss-experimental-show-pixel-values '()
+  ""
+  :type '(repeat string)
+  :package-version '(lsp-mode . "7.1"))
+
+(lsp-register-custom-settings
+ '(("tailwindCSS.includeLanguages" lsp-tailwindcss-include-languages)
+   ("tailwindCSS.emmetCompletions" lsp-tailwindcss-emmet-completions)
+   ("tailwindCSS.colorDecorators" lsp-tailwindcss-color-decorators)
+   ("tailwindCSS.validate" lsp-tailwindcss-validate)
+   ("tailwindCSS.lint.cssConflict" lsp-tailwindcss-lint-css-conflict)
+   ("tailwindCSS.lint.invalidApply" lsp-tailwindcss-lint-invalid-apply)
+   ("tailwindCSS.lint.invalidScreen" lsp-tailwindcss-lint-invalid-screen)
+   ("tailwindCSS.lint.invalidVariant" lsp-tailwindcss-lint-invalid-variant)
+   ("tailwindCSS.lint.invalidConfigPath" lsp-tailwindcss-lint-invalid-config-path)
+   ("tailwindCSS.lint.invalidTailwindDirective" lsp-tailwindcss-lint-invalid-tailwind-directive)
+   ("tailwindCSS.experimental.classRegex" lsp-tailwindcss-experimental-class-regex)
+   ("tailwindCSS.experimental.showPixelValues" lsp-tailwindcss-experimental-show-pixel-values)))
+
+(defun lsp-tailwindcss--config-updated ()
+  ;; FIXME: Update decorations in editor, see original implementation
+  ;; https://github.com/tailwindlabs/tailwindcss-intellisense/blob/b3c17cf5c9d1d74091a4a5ee0504696bb2ce1c46/packages/tailwindcss-intellisense/src/lib/registerColorDecorator.ts#L125
+  t)
+
+(defun lsp-tailwindcss--config-error ()
+  ;; FIXME: Update decorations in editor and show error message, see original implementation
+  ;; https://github.com/tailwindlabs/tailwindcss-intellisense/blob/b3c17cf5c9d1d74091a4a5ee0504696bb2ce1c46/packages/tailwindcss-intellisense/src/lib/registerColorDecorator.ts#L129
+  ;; https://github.com/tailwindlabs/tailwindcss-intellisense/blob/b3c17cf5c9d1d74091a4a5ee0504696bb2ce1c46/packages/tailwindcss-intellisense/src/lib/registerConfigErrorHandler.ts#L5
```

Comment:
for these you will have to port updateDecorationsInEditor - here I need more specific questions because it is a huge method

Reply:
`updateDecorationInEditor` provides syntax highlight (see the [doc](https://tailwindcss.com/docs/intellisense)).
Is lsp-mode support syntax highlighting?

Reply:
We do syntax highlighting via sematic tokens. This looks more like textDocument/documentColor handling.

Reply:
So it's already implemented and I have nothing to do about this?